full-stack-tests/backend/src/integration/CloudWorkspace.test.ts

```diff
@@ -7,156 +7,278 @@ import "./StartupShutdown"; // calls startup/shutdown IModelHost before/after al
-    expect(wsCont2.resolveDbFileName({ dbName: testDbName, version: "~1.1.0" })).contains("1.1.3");
-    expect(wsCont2.resolveDbFileName({ dbName: testDbName, version: "1.2.0" })).contains("1.2.0");
-    expect(wsCont2.resolveDbFileName({ dbName: testDbName, version: ">1.0.0 <3.0.0" })).contains("1.2.4");
-    expect(wsCont2.resolveDbFileName({ dbName: testDbName, version: "1.0.0" })).contains("1.0.0");
-
-    ws2 = wsCont2.getWorkspaceDb({ dbName: testDbName, version: "~1.1.0", includePrerelease: true });
-    expect(ws2.dbFileName).contains("1.1.4-beta");
-    expect(ws2.getString("string 1")).equals(newVal);
-    expect(ws2.getString("myVersion")).equals("1.1.4-beta");
-    ws2.container.closeWorkspaceDb(ws2);
-
-    ws2 = wsCont2.getWorkspaceDb({ dbName: testDbName });
-    expect(ws2.dbFileName).contains("3.0.0");
-    expect(ws2.getString("string 1")).equals("value of string 1");
-    expect(ws2.getString("myVersion")).equals("3.0.0");
-    ws2.container.closeWorkspaceDb(ws2);
-
-    workspace1.close();
-    workspace2.close();
-
-    const dict = {
-      "cloud/containers": [
-        {
-          name: "test/container1",
-          baseUri: "http://127.0.0.1:10000/devstoreaccount1",
-          storageType: "azure",
-          containerId,
-        },
-      ],
-      "workspace/databases": [
-        {
-          name: "test/test1",
-          dbName: testDbName,
-          containerName: "test/container1",
-          version: "^1",
-        },
-      ],
+    const defineStyles = (editDb: Workspace.Editor.EditableDb, prefix: string, bias: number) => {
+      for (let i = 0; i < 10; ++i) {
+        const styleName = `styles/num-${i}`;
+        editDb.updateString(styleName, `${prefix}/value ${i}`);
+        editDb.updateBlob(styleName, new Uint8Array([i + bias]));
+      }
     };
-    const workspace3 = Workspace.construct(new BaseSettings(), { containerDir: join(IModelHost.cacheDir, "TestWorkspace3"), testCloudCache: makeCloudCache("test3") });
-    workspace3.settings.addDictionary("testDict", SettingsPriority.application, dict);
-    const db = await workspace3.getWorkspaceDb("test/test1");
-    expect(db.dbFileName).equal("testDb:1.2.4");
-    expect(db.dbName).equal(testDbName);
-    expect(db.container.id).equal(containerId);
 
-    const db2 = await workspace3.getWorkspaceDb("test/test1");
-    expect(db2).equal(db);
+    await withPatchVersion(styles1, (editDb) => {
+      defineStyles(editDb, "batch1", 100);
+    });
+    await withPatchVersion(styles2, (editDb) => {
+      defineStyles(editDb, "batch2", 200);
+    });
+
+    const dbs: WorkspaceDb.CloudProps[] = [
+      { ...styles1.cloudProps!, loadingHelp: "if problems with styles1", prefetch: true },
+      { ...styles1.cloudProps!, containerId: "styles3", loadingHelp: "if problems with styles3" },
+      { ...styles2.cloudProps!, loadingHelp: "if problems with styles2" },
+      { ...styles1.cloudProps!, loadingHelp: "if problems with styles1", prefetch: true },
+    ];
```

Comment:
In practice, were would this list of Cloud WorkspaceDbs come from? iModelSettings?

Reply:
please see latest